unstructured/ingest/pipeline/reformat/chunking.py

```diff
@@ -62,3 +85,51 @@ def run(self, elements_json: str) -> Optional[str]:
 
     def get_path(self) -> Path:
         return (Path(self.pipeline_context.work_dir) / "chunked").resolve()
+
+    def chunk(self, elements_json_file: str) -> Optional[list[Element]]:
+        """Called by Chunker.run() to properly execute the defined chunking_strategy."""
+        # -- No chunking_strategy means no chunking --
+        if self.chunking_config.chunking_strategy is None:
+            return
+        # -- Chunk locally --
+        if self.chunking_config.chunking_strategy in ("basic", "by_title"):
+            return dispatch.chunk(
+                elements=elements_from_json(filename=elements_json_file),
+                chunking_strategy=self.chunking_config.chunking_strategy,
+                combine_text_under_n_chars=self.chunking_config.combine_text_under_n_chars,
+                include_orig_elements=self.chunking_config.include_orig_elements,
+                max_characters=self.chunking_config.max_characters,
+                multipage_sections=self.chunking_config.multipage_sections,
+                new_after_n_chars=self.chunking_config.new_after_n_chars,
+                overlap=self.chunking_config.overlap,
+                overlap_all=self.chunking_config.overlap_all,
+            )
+        # -- Chunk remotely --
+        if self.partition_config.partition_by_api:
+            return partition_via_api(
+                filename=elements_json_file,
+                # -- If api_key or api_url are None, partition_via_api will raise an error, which
+                # -- will be caught and logged by Chunker.run()
+                api_key=self.partition_config.api_key,  # type: ignore
+                api_url=self.partition_config.partition_endpoint,  # type: ignore
+                chunking_strategy=self.chunking_config.chunking_strategy,
+                combine_under_n_chars=self.chunking_config.combine_text_under_n_chars,
+                # -- These are not currently supported by the sdk and raise an error.
+                # -- They will need to be updated once we bump the client version.
+                # combine_text_under_n_chars=self.chunking_config.combine_text_under_n_chars,
+                # include_orig_elements=self.chunking_config.include_orig_elements,
+                max_characters=self.chunking_config.max_characters,
+                multipage_sections=self.chunking_config.multipage_sections,
+                new_after_n_chars=self.chunking_config.new_after_n_chars,
+                # -- These are not supported by older versions of unstructured-client, which we have
+                # -- pinned to <=0.18.0
+                # overlap=self.chunking_config.overlap,
+                # overlap_all=self.chunking_config.overlap_all,
+            )
```

Comment:
What is the behavior when there is no chunking strategy registered with the requested name on the remote host?

Like how does the user get a message like: `No such chunking strategy: 'foobar'`?